spring-data-jdbc/src/main/java/org/springframework/data/jdbc/core/JdbcAggregateTemplate.java

```diff
@@ -425,6 +428,12 @@ public <T> void deleteAll(Iterable<? extends T> instances) {
 		}
 	}
 
+	private <T> void verifyIdProperty(T instance) {
+
+		Class<?> type = instance.getClass();
+		Assert.isTrue(context.getRequiredPersistentEntity(type).hasIdProperty(),() -> "Aggregate root must have an id property. " + type.getName() + " has none");
```

Comment:
Nit: `String.format(…)`

Also, I am not sure that `IllegalArgumentException` is the right exception type. `PersistentEntity.getRequiredIdProperty()` uses `IllegalStateException`, `MappingException` could be an alternative.

Generally, why don't we use `PersistentEntity.getRequiredIdProperty()`?

Reply:
I switched back an forth between various variants, But now go with the simple `PeristentEntity.getRequiredIdProperty()` although it is a little less expressive.